polyfill/test/ecmascript.mjs

```diff
@@ -64,8 +64,644 @@ describe('ECMAScript', () => {
       });
     }
   });
+
+  describe('GetAvailableNamedTimeZoneIdentifier', () => {
+    it('Case-normalizes time zone IDs', () => {
+      const ids = [...new Set([...timeZoneIdentifiers, ...Intl.supportedValuesOf('timeZone')])];
+      for (const id of ids) {
+        const lower = id.toLowerCase();
+        const upper = id.toUpperCase();
+        equal(ES.GetAvailableNamedTimeZoneIdentifier(id)?.identifier, id);
+        equal(ES.GetAvailableNamedTimeZoneIdentifier(upper)?.identifier, id);
+        equal(ES.GetAvailableNamedTimeZoneIdentifier(lower)?.identifier, id);
+      }
+    });
+    it('Returns canonical IDs', () => {
+      const ids = Intl.supportedValuesOf('timeZone');
+      for (const id of ids) {
+        equal(ES.GetAvailableNamedTimeZoneIdentifier(id).primaryIdentifier, id);
+      }
+      const knownAliases = [
+        ['America/Atka', 'America/Adak'],
+        ['America/Knox_IN', 'America/Indiana/Knox'],
+        ['Asia/Ashkhabad', 'Asia/Ashgabat'],
+        ['Asia/Dacca', 'Asia/Dhaka'],
+        ['Asia/Istanbul', 'Europe/Istanbul'],
+        ['Asia/Macao', 'Asia/Macau'],
+        ['Asia/Thimbu', 'Asia/Thimphu'],
+        ['Asia/Ujung_Pandang', 'Asia/Makassar'],
+        ['Asia/Ulan_Bator', 'Asia/Ulaanbaatar']
+      ];
+      for (const [identifier, primaryIdentifier] of knownAliases) {
+        const record = ES.GetAvailableNamedTimeZoneIdentifier(identifier);
+        equal(record.identifier, identifier);
+        equal(record.primaryIdentifier, primaryIdentifier);
+      }
+    });
+  });
 });
 
+const timeZoneIdentifiers = [
+  'Africa/Abidjan',
+  'Africa/Algiers',
+  'Africa/Bissau',
+  'Africa/Cairo',
+  'Africa/Casablanca',
```

Comment:
Unfortunately that kind of authoritative source doesn't really exist for TZDB because of the multitude of build options. And even if it did, all the engines use CLDR as their data source so knowing the TZDB data isn't actually helpful because if CLDR doesn't include it then the engines throw anyways. 

But I think your general point is a good one: can we attribute this data to a source and make it easier to refresh?

I updated from using a static array to using a `curl`-ed  copy of CLDR's JSON data file, and added comments explaining how to refresh and parse it. The net result is the same: a list of IDs, but hopefully with more clarity about how it's generated.

Better?  Let me know if you think we need more than this.

Reply:
No, that's great. Thank you.